src/GafferImageUI/ImageView.cpp

```diff
@@ -254,6 +276,9 @@ ImageView::ImageView( const std::string &name )
 	viewportGadget()->setPrimaryChild( m_imageGadget );
 
 	m_colorInspector = shared_ptr<ColorInspector>( new ColorInspector( this ) );
+
+	m_zoomLevelChangedConnection = zoomLevelChangedSignal().connect( boost::bind( &ImageView::setZoomLevel, this, ::_1) );
```

Comment:
Take it up with boost then! I think you'll find though that the answer is that the maker of the connection should be the only one in a position to modify the connection afterwards - allowing outside code to modify _your_ connection would break your code, so they're actually protecting you.